tests/functional/events/09-task-event-mail.t

```diff
@@ -49,15 +49,16 @@ run_ok "${TEST_NAME_BASE}-validate" \
 workflow_run_ok "${TEST_NAME_BASE}-run" \
     cylc play --reference-test --debug --no-detach ${OPT_SET} "${WORKFLOW_NAME}"
 
-contains_ok "${TEST_SMTPD_LOG}" <<__LOG__
-b'retry: 1/t1/01'
-b'succeeded: 1/t1/02'
-b'see: http://localhost/stuff/${USER}/${WORKFLOW_NAME}/'
-__LOG__
+grep_ok 'retry: 1/t1/01' "${TEST_SMTPD_LOG}"
+grep_ok 'succeeded: 1/t1/02' "${TEST_SMTPD_LOG}"
+grep_ok "see: http://localhost/stuff/${USER}/${WORKFLOW_NAME}/" "${TEST_SMTPD_LOG}"
+
+cat "${TEST_SMTPD_LOG}" > /home/h02/tpilling/foo
+
 run_ok "${TEST_NAME_BASE}-grep-log" \
-    grep -q "Subject: \\[1/t1/01 retry\\].* ${WORKFLOW_NAME}" "${TEST_SMTPD_LOG}"
+    grep -q "Subject: \\[1/t1/01 retry\\]" "${TEST_SMTPD_LOG}"
```

Comment:
Annoyingly the workflow name is no longer on the same line. I don't think it's necessary but can come up with an alternative approach. 

n.b. Using Perl style regex (`-P`) and searching for a multiline is horribly slow.

Reply:
> Annoyingly the workflow name is no longer on the same line.

That's not related to this change / Python 3.12 is it?

If not, why is this test only failing now?

Reply:
Because the output from `aiosmtpd` is different than that from `smtpd`.

Reply:
(note, I've recently had to deal with the same issue in these same tests in https://github.com/cylc/cylc-flow/pull/5769, I used `grep -P` and found it to be ok I think).

Reply:
I tried that and found it painful.

Reply:
It's fine. Will put it back.